tests/integration/test_integration.py

```diff
@@ -234,37 +235,95 @@ def test_pex_repl_built():
         assert b">>>" in stdout
 
 
-@pytest.mark.skipif(
-    IS_PYPY or IS_MAC,
-    reason="REPL history is only supported on CPython. It works on macOS in an interactive "
```

Comment:
This appears to never have been true. In fact, using `PEX_INTERPRETER_HISTORY` / `PEX_INTERPRETER_HISTORY_FILE` on any Python interpreter without readline support, even CPython, previously led to:
```
$ PEX_INTERPRETER_HISTORY=1 PEX_INTERPRETER_HISTORY_FILE=/tmp/check $PWD/install/bin/python3.12 empty.pex
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/__main__.py", line 106, in <module>
    bootstrap_pex(__entry_point__, execute=__execute__, venv_dir=__venv_dir__)
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex_bootstrapper.py", line 627, in bootstrap_pex
    pex.PEX(entry_point).execute()
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex.py", line 561, in execute
    sys.exit(self._wrap_coverage(self._wrap_profiling, self._execute))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex.py", line 468, in _wrap_coverage
    return runner(*args)
           ^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex.py", line 499, in _wrap_profiling
    return runner(*args)
           ^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex.py", line 585, in _execute
    return self.execute_interpreter()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.pex/unzipped_pexes/e2c4c91895f250a4f203b1d575cb3c39208ea5a7/.bootstrap/pex/pex.py", line 661, in execute_interpreter
    import readline
ModuleNotFoundError: No module named 'readline'
```
(Here I hand-built CPython3.12 with `./configure --prefix $PWD/install --without-readline`)

Reply:
Hmm, archaeology on #2018 (whose CI logs have unfortunately expired) finds a commit message stating that the issue on pypy (in CI) was apparently related to terminfo, so may have been due to interaction between pypy and the CI environment rather than pypy per se, but AFAICT wasn't due to unavailability of readline. Similarly, the macos ENOTTY issue only manifested in CI, and was not related to the presence of readline.

Anyway, if this is now deemed false and the actual underlying issue is just readline then the mentions of CPython in pex/variables.py#531 and pex/variables.py#542 should be removed.

Reply:
Yeah, definitely false. CI passes for `pypy3.10` and mac and I've personally tested for `pypy{2.7,3.{5,6,7,8,9,10}}`. This PR had to plumb `TERM` to work around 1/2 of the tty issues (using pexpect / ptys was the other half).

Variables docs fixed.

Reply:
I should have dived deeper in #2018 then, sounds like TERM would have made things work.